src/main/webapp/app/exercises/modeling/assess/modeling-assessment.component.ts

```diff
@@ -270,9 +281,27 @@ export class ModelingAssessmentComponent implements AfterViewInit, OnDestroy, On
             elementType: feedback.referenceType! as UMLElementType | UMLRelationshipType,
             score: feedback.credits!,
             feedback: feedback.text || undefined,
+            label: this.calculateLabel(feedback),
+            labelColor: this.calculateLabelColor(feedback),
         }));
         if (this.apollonEditor) {
             this.apollonEditor!.model = this.model;
         }
     }
+
+    private calculateLabel(feedback: any) {
+        if (this.highlightDifferences && this.languageService.getCurrentLanguage() === 'en') {
+            return feedback.copiedFeedbackId ? 'First correction round' : 'Second correction round';
+        } else if (this.highlightDifferences && this.languageService.getCurrentLanguage() === 'de') {
+            return feedback.copiedFeedbackId ? 'Erste Korrekturrunde' : 'Zweite Korrekturrunde';
+        }
+        return undefined;
+    }
+
+    private calculateLabelColor(feedback: any) {
+        if (this.highlightDifferences) {
+            return feedback.copiedFeedbackId ? '#3e8acc' : '#ffa561';
```

Comment:
The two colors seem to be instances of hardcoded magic numbers. 
Wouldn't it be better to define two constants like `const firstCorrectionRoundColor = #3e8acc`, etc., or even create new (global) `enums` so you could reuse them in your test class as well?

Or is it common in Artemis to use colors like magic numbers?

Reply:
Good finding, I refactored this and moved all definitions into class variables